### ISSUE TITLE
Docs and crate publishing

### ISSUE THREAD

**Issue by halvnykterist** [hussein-aitlahcen/gns-rs#1]
This is really cool! Is there a reason it hasn't been published as a crate? Having the autogenerated docs readily available would help a lot.

**Comment by hussein-aitlahcen** [comment]
The crate is not compilling right now (it does for me because I already applied some patchs to GNS), I'm waiting for https://github.com/ValveSoftware/GameNetworkingSockets/pull/248 and https://github.com/ValveSoftware/GameNetworkingSockets/pull/251 to land, then setup CI for Windows and Linux build.

**Comment by hussein-aitlahcen** [comment]
@halvnykterist I've decided to use my fork in the meantime, the CI is passing for both windows and linux.
I've also added more rustdoc and will publish the crate.

**Comment by hussein-aitlahcen** [comment]
Unfortunately, crates.io does not support submodules... The maximum upload size is 10mb and the submodule is too big to fit. I am not going to tweak the build.rs to handle this. For the documentation, we'll use github pages for now. [The doc is available here](https://hussein-aitlahcen.github.io/gns-rs/gns/)

**Comment by halvnykterist** [comment]
Shame about the submodules, but the docs are really what's important anyway. Excited to check it out.

**Comment by 2l47** [comment]
@hussein-aitlahcen With those PRs being closed, is the crate ready to be published?

**Comment by hussein-aitlahcen** [comment]
> @hussein-aitlahcen With those PRs being closed, is the crate ready to be published?

Yes I think we can push an initial version. Was awaiting an upstream PR to be merged.

**Comment by hussein-aitlahcen** [comment]
@2l47 sadly the crate can't be published due to the above limitation. If you want to use it you can reference it in cargo using a git dependency.

**Comment by halotroop2288** [comment]
>  in cargo using a git dependency

```toml
[dependencies]
gns = { git = "https://github.com/hussein-aitlahcen/gns-rs", branch = "master" }
```
This results in the error:
```
error: failed to run custom build command for `gns-sys v0.1.0 (https://github.com/hussein-aitlahcen/gns-rs?branch=master#8daa3561)`
```

So, I don't think that's going to work.

**Comment by hussein-aitlahcen** [comment]
> > in cargo using a git dependency
> 
> ```toml
> [dependencies]
> gns = { git = "https://github.com/hussein-aitlahcen/gns-rs", branch = "master" }
> ```
> 
> This results in the error:
> 
> ```
> error: failed to run custom build command for `gns-sys v0.1.0 (https://github.com/hussein-aitlahcen/gns-rs?branch=master#8daa3561)`
> ```
> 
> So, I don't think that's going to work.

Hi, could you provide the full trace please ? Beware of clang required to build the valve lib

**Comment by halotroop2288** [comment]
> Hi, could you provide the full trace please ?

Here it is in a test project, to isolate any other issues.
https://hastebin.com/tehorogovu.less

**Comment by hussein-aitlahcen** [comment]
> > Hi, could you provide the full trace please ?
> 
> Here it is in a test project, to isolate any other issues. https://hastebin.com/tehorogovu.less

Looks like you are missing the protobuf compiler in your path.

**Comment by debroejm** [comment]
Hmm, that would work, and is an option, but effectively makes offline builds not possible using repository mirrors. But then again, we've already made that impossible on Windows due to needing to clone vcpkg...
